libnavigation-ui/src/main/java/com/mapbox/navigation/ui/routealert/MapboxRouteAlert.kt

```diff
@@ -0,0 +1,121 @@
+ * @param options combination of [RouteAlertOption]s to indicate which type of route alerts
+ * should be handled
+ *  e.g.
+ *  options = [RouteAlertOption.ROUTE_ALERT_REST_STOP] or [RouteAlertOption.ROUTE_ALERT_TUNNEL]
+ *  to support both rest stop and tunnel.
+ */
+class MapboxRouteAlert @JvmOverloads constructor(
+    context: Context,
+    style: Style,
+    private val options: Int = Int.MAX_VALUE
+) {
+
+    private val routeAlertToll = RouteAlertToll(
+        RouteAlertViewOptions.Builder(context, style).build()
+    )
+
+    /**
+     * Display supported [RouteAlert] on the map.
+     * Which types of [RouteAlert] are supported relies on the [options] value. Only supported
+     * [RouteAlert] can be handle and displayed on the map.
+     *
+     * @param routeAlerts a list of route alerts
+     */
+    fun onNewRouteAlerts(routeAlerts: List<RouteAlert>) {
+        if (options and RouteAlertOption.ROUTE_ALERT_TOLL
+            == RouteAlertOption.ROUTE_ALERT_TOLL
+        ) {
+            routeAlertToll.onNewRouteAlerts(routeAlerts)
+        }
+    }
+
+    companion object {
+        /**
+         * Mapbox pre-defined properties line type for route alert.
+         * @see [RouteAlertOption.ROUTE_ALERT_TUNNEL], [RouteAlertOption.ROUTE_ALERT_RESTRICTED_AREA]
+         */
+        fun getMapboxRouteAlertLineLayerProperties(
+            @ColorInt color: Int,
+            width: Float
+        ): Array<PropertyValue<out Any>> =
+            arrayOf(
+                PropertyFactory.lineColor(color),
+                PropertyFactory.lineWidth(width)
+            )
+
+        /**
+         * Mapbox pre-defined symbol layer properties for route alert.
+         * @see [RouteAlertOption.ROUTE_ALERT_COUNTRY_BORDER_CROSSING]
+         * [RouteAlertOption.ROUTE_ALERT_REST_STOP]
+         * [RouteAlertOption.ROUTE_ALERT_TOLL]
+         */
+        fun getMapboxRouteAlertSymbolLayerProperties(): Array<PropertyValue<out Any>> = arrayOf(
+            PropertyFactory.iconSize(1.5f),
+            PropertyFactory.iconAnchor(ICON_ANCHOR_CENTER),
+            PropertyFactory.textAnchor(Property.TEXT_ANCHOR_TOP),
+            PropertyFactory.iconAllowOverlap(true),
+            PropertyFactory.iconIgnorePlacement(true)
+        )
+
+        /**
+         * Mapbox pre-defined symbol layer properties for Tunnel name
+         * @see [RouteAlertOption.ROUTE_ALERT_TUNNEL]
+         */
+        fun getMapboxTunnelNameLayerProperties(): Array<PropertyValue<out Any>> = arrayOf(
+            PropertyFactory.iconSize(2f),
+            PropertyFactory.iconAnchor(ICON_ANCHOR_CENTER),
+            PropertyFactory.textAnchor(Property.TEXT_ANCHOR_BOTTOM_RIGHT),
+            PropertyFactory.iconAllowOverlap(true),
+            PropertyFactory.iconIgnorePlacement(true),
+            PropertyFactory.textAllowOverlap(true),
+            PropertyFactory.textIgnorePlacement(true)
+        )
+    }
+}
+
+/**
+ * Different route alert options
+ */
+object RouteAlertOption {
```

Comment:
This lists a lot of types, while only tolls are actually supported. In any case, I'd suggest a builder style pattern anyway.

Reply:
removed.